### ISSUE TITLE
Preprocessing of training data

### ISSUE THREAD

**Issue by XiaoHeng999** [Kunbyte-AI/ROSE#14]
This is a very excellent piece of work. I would like to ask if you could share the code for data preprocessing during your training phase? Specifically, the part of the code where you subtract the removed video from the original video to get the mask!This is because the mask I got doesn't seem to be very ideal.Thank！

**Comment by m1a0cX** [comment]
Thank you for your attention. Actually I don't quite figure out what do you mean. The mask video, unedited video and edited video are all rendered by Unreal Engine and during training they all will be processed into 720P with 90 frames. And the ground-truth difference mask is derived from subtracting the edited video with unedited video and then binarizing the subtracted result. Do you mean the rendered videos or the ground-truth difference masks are not in very high quality?

**Comment by XiaoHeng999** [comment]
<img width="765" height="708" alt="Image" src="https://github.com/user-attachments/assets/37b24b33-e561-451a-9dec-5ebf6bbcda14" />
As shown in the image, in the pipeline of the paper here, the portion obtained by subtracting the Original Unedited Video from the Edited Video to get the Ground-Truth Difference Mask may not be very ideal in my case. I am not sure how you did it, and I have some doubts about this part. If it's convenient, could you please explain it to me? If possible, could you also provide me with the code for this part?

**Comment by XiaoHeng999** [comment]
data process of training ：<img width="1154" height="483" alt="Image" src="https://github.com/user-attachments/assets/eba4d9bd-db4d-48fc-879b-aefc122a5ef7" />  infer：

<img width="1472" height="570" alt="Image" src="https://github.com/user-attachments/assets/3368710d-e750-4078-a368-806e93359b8e" />

While studying your code, I found another issue and hope to get some clarification. In the data processing part during the training phase, the mask is multiplied with the video (as shown in the code in Figure 1). However, in the inference phase, you do not perform this multiplication, and the input video is used directly (as shown in the code in Figure 2). I am a bit confused about this. I am not sure if you process this elsewhere in the training script, and if so, could you provide me with a brief explanation? Thank you!

**Comment by m1a0cX** [comment]
> data process of training ：<img alt="Image" width="1154" height="483" src="https://private-user-images.githubusercontent.com/237510771/543293105-eba4d9bd-db4d-48fc-879b-aefc122a5ef7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk5MTY5MDgsIm5iZiI6MTc2OTkxNjYwOCwicGF0aCI6Ii8yMzc1MTA3NzEvNTQzMjkzMTA1LWViYTRkOWJkLWRiNGQtNDhmYy04NzliLWFlZmMxMjJhNWVmNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIwMVQwMzMwMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZGVkNDNlNzZmM2RmNjcxMWQ3OGE5NmMxMjljZGRiZDYwYmE0NjQwM2ZkZTM3MGNiODY4ZmQ2NWVjNWMxNTUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.axj_Bj0iLnWYUf_xjbsa3kYOcTTf0VWO4JkDgS1Q8qI"> infer：
> 
> <img alt="Image" width="1472" height="570" src="https://private-user-images.githubusercontent.com/237510771/543293780-3368710d-e750-4078-a368-806e93359b8e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk5MTY5MDgsIm5iZiI6MTc2OTkxNjYwOCwicGF0aCI6Ii8yMzc1MTA3NzEvNTQzMjkzNzgwLTMzNjg3MTBkLWU3NTAtNDA3OC1hMzY4LTgwNmU5MzM1OWI4ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIwMVQwMzMwMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NGU2MmU3OTI2YjJjM2MwZGExMmY0Y2YwMTg2M2E0ZjEzM2E4YzUwZWRhODVhODE0ZGM3OTgzZjAxZjgxMWQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.rTrkc7MKDDAptoqP0tsvGicSS-7E92N5eFKcvWhYQ9M">
> While studying your code, I found another issue and hope to get some clarification. In the data processing part during the training phase, the mask is multiplied with the video (as shown in the code in Figure 1). However, in the inference phase, you do not perform this multiplication, and the input video is used directly (as shown in the code in Figure 2). I am a bit confused about this. I am not sure if you process this elsewhere in the training script, and if so, could you provide me with a brief explanation? Thank you!

I think you may have misunderstood. We don't use the first code snippet you pasted. And our related training snippet is like this: 

<img width="1123" height="929" alt="Image" src="https://github.com/user-attachments/assets/d3d0deda-6db3-4cfa-b2f6-c8d1545698dd" />

So I think it aligns with our inference code.

As for getting the dataset, our code is like this:

<img width="995" height="962" alt="Image" src="https://github.com/user-attachments/assets/1fe526fe-0d02-4746-8d8b-fa952b388d18" />

And when creating the dataset class we will use a self-defined collate-fn function:

<img width="896" height="444" alt="Image" src="https://github.com/user-attachments/assets/6de0012a-69f7-4518-a85c-2e63a4f79918" />

**Comment by m1a0cX** [comment]
SInce our code is based on the [Wan2.1-Fun-1.3B-Inpaint](https://github.com/aigc-apps/VideoX-Fun), there may be some irrelevant code left. We're sorry if that bothers you.

**Comment by XiaoHeng999** [comment]
<img width="507" height="717" alt="Image" src="https://github.com/user-attachments/assets/9511a154-3bc9-401d-8722-5cc9216fe17c" />Thank you for your reply. The diff_mask I obtained may have some noise. Is your result like this? @m1a0cX

**Comment by m1a0cX** [comment]
> <img alt="Image" width="507" height="717" src="https://private-user-images.githubusercontent.com/237510771/543377598-9511a154-3bc9-401d-8722-5cc9216fe17c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk5MzA5OTAsIm5iZiI6MTc2OTkzMDY5MCwicGF0aCI6Ii8yMzc1MTA3NzEvNTQzMzc3NTk4LTk1MTFhMTU0LTNiYzktNDAxZC04NzIyLTVjYzkyMTZmZTE3Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIwMVQwNzI0NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yODljODhlM2RkZjU3NmI5ZDZhZWUxODM5NGUyZDMxMjRjMDk4YjU3OGU3OWEwYWNjMzk2OGY4MmEzOWJkMjA1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Re_PSZNNHrmzZN9FVWaSMSWEQ5oB-55TbsjfMGIzsXQ">Thank you for your reply. The diff_mask I obtained may have some noise. Is your result like this? [@m1a0cX](https://github.com/m1a0cX)

Yep. I think it's related to the setting of difference threshold. Also I think there is still room for improvement of the current supervision module(difference mask predictor).
